### PR TITLE
Implemented JWT token

### PR DESCRIPTION
**Pull request by vikkushnir**

### CODE REVIEW

File: src/main/java/online/book/store/service/user/UserAuthenticationServiceImpl.java [hunk 1]
```diff
@@ -37,4 +45,15 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto) {
...
+
+    @Override
+    public UserLoginResponseDto authenticate(UserLoginRequestDto requestDto) {
+        final Authentication authentication = authenticationManager.authenticate(
+                new UsernamePasswordAuthenticationToken(
+                        requestDto.email(), requestDto.password())
+        );
+
+        String token = jwtUtil.generateToken(authentication.getName());
+        return new UserLoginResponseDto(token);
```
Comment (d1sam) [inline src/main/java/online/book/store/service/user/UserAuthenticationServiceImpl.java]: let's better create `AuthenticationService` with the method `authenticate`

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -7,3 +7,6 @@ spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.show-sql=true
 spring.jpa.open-in-view=false
+
+jwt.expiration=500000
+jwt.secret=helloMatesHereIsSecretMessageAbracadabra
```
Comment (d1sam) [inline src/main/resources/application.properties]: let's provide these values from env properties

File: src/main/java/online/book/store/service/user/UserAuthenticationService.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+import online.book.store.dto.request.user.UserLoginRequestDto;
+import online.book.store.dto.request.user.UserRegistrationRequestDto;
+import online.book.store.dto.response.user.UserLoginResponseDto;
+import online.book.store.dto.response.user.UserResponseDto;
+
+public interface UserAuthenticationService {
+    UserResponseDto register(UserRegistrationRequestDto requestDto);
+
+    UserLoginResponseDto authenticate(UserLoginRequestDto requestDto);
```
Comment (d1sam) [inline src/main/java/online/book/store/service/user/UserAuthenticationService.java]: according to SRP and IS principles of SOLID you should create separate interface for authentication purposes.

File: src/main/java/online/book/store/dto/request/user/UserLoginRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+
+public record UserLoginRequestDto(
+        @NotBlank
+        @Email
+        String email,
+        @NotBlank
+        @Size(min = 8)
```
Comment (d1sam) [inline src/main/java/online/book/store/dto/request/user/UserLoginRequestDto.java]: what about max param?

File: src/main/java/online/book/store/dto/request/user/UserLoginRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package online.book.store.dto.request.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+
+public record UserLoginRequestDto(
+        @NotBlank
+        @Email
+        String email,
```
Comment (d1sam) [inline src/main/java/online/book/store/dto/request/user/UserLoginRequestDto.java:11]: what about the size?

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but there are some minor changes, that should be done according my comments.

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**